### PR TITLE
Type safe query params

### PR DESCRIPTION
**Pull request by vasily-kartashov**

### CODE REVIEW

File: frameworks/PHP/hamlet/hamlet.dockerfile [hunk 1]
```diff
@@ -1,22 +1,22 @@
-FROM ubuntu:18.10
+FROM debian:stretch
```
Comment (joanhey) [inline frameworks/PHP/hamlet/hamlet.dockerfile]: Any reason to use debian:stretch ??

I's easier use the same when we make changes and also all use the same stack in the bench.

Comment (vasily-kartashov) [reply frameworks/PHP/hamlet/hamlet.dockerfile]: @joanhey revereted back to ubuntu

Comment (joanhey) [reply frameworks/PHP/hamlet/hamlet.dockerfile]: Thank you.
Now it's better for the docker cache.
Also debian stretch have nginx 1.10.3.

And the next week we will change to ubuntu 19.04, that come with nginx 1.15.9.
From 1.15.6 have fastcgi_socket_keepalive, and we will try it.

**Comment by nbrady-techempower** [comment]
Unsure why this is showing 44 files changed against master. I'm going to close and reopen and see if that helps so it's easier to review.

**Comment by nbrady-techempower** [comment]
That, of course, did nothing. @valyala Would you mind rebasing this or opening it as a new PR?

File: frameworks/PHP/hamlet/benchmark_config.json [hunk 2]
```diff
@@ -47,26 +46,22 @@
       "notes": "",
       "versus": "php"
```
Comment (joanhey) [inline frameworks/PHP/hamlet/benchmark_config.json]: Please change to:
`"versus": "swoole"`

So we will see the framework overhead correctly.
Check:
https://www.techempower.com/benchmarks/#section=test&runid=9f738614-61d5-4c04-a1a5-3778b2ac3003&hw=ph&test=fortune&l=zik073-7
and click the tab `Framework overhead`

File: frameworks/PHP/hamlet/benchmark_config.json [hunk 3]
```diff
@@ -47,26 +46,22 @@
...
-      "webserver": "react",
+      "orm": "micro",
+      "platform": ".NET",
+      "webserver": "kestrel",
       "os": "Linux",
       "database_os": "Linux",
-      "display_name": "hamlet-react",
+      "display_name": "hamlet-peachpie",
       "notes": "",
       "versus": "php"
```
Comment (joanhey) [inline frameworks/PHP/hamlet/benchmark_config.json]: Please change to:
"versus": "peachpie"

Comment (joanhey) [reply frameworks/PHP/hamlet/benchmark_config.json]: Please this one too.

**Pull request closed without merging**